### PR TITLE
Add unit tests for FirebaseCore

### PR DESCRIPTION
**Pull request by bstpierr**
Added several unit tests to help increase the overall coverage on Firebase Core

### CODE REVIEW

File: Example/Core/Tests/FIRReachabilityCheckerTest.m [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+  XCTAssertEqual(FakeReachability.scheduleCall, 3, @"");
+  XCTAssertEqual(FakeReachability.unscheduleCall, -1, @"");
+  XCTAssertEqual(FakeReachability.releaseCall, 4, @"");
+
+  XCTAssertEqual([statuses_ count], (NSUInteger)0, @"");
+}
+
+- (void)testBadHost {
+  XCTAssertNil([[FIRReachabilityChecker alloc] initWithReachabilityDelegate:self
+                                                             loggerDelegate:self
```
Comment (paulb777) [inline Example/Core/Tests/FIRReachabilityCheckerTest.m]: Fix build warning

File: Example/Core/Tests/FIRReachabilityCheckerTest.m [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+                                                                   withHost:nil],
+               @"Creating a checker with nil hostname must fail.");
+  XCTAssertNil([[FIRReachabilityChecker alloc] initWithReachabilityDelegate:self
+                                                             loggerDelegate:self
```
Comment (paulb777) [inline Example/Core/Tests/FIRReachabilityCheckerTest.m]: Fix build warning

File: Example/Core/Tests/GTMHTTPServer.h [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+//  not meant to stand up a site, it sends all requests to its delegate for
+//  processing on the main thread.  It does not support pipelining, etc.  It's
+//  great for places where you need a simple webserver to unittest some code
+//  that hits a server.
+//
+//  NOTE: there are several TODOs left in here as markers for things that could
+//  be done if one wanted to add more to this class.
+//
+//  Based a little on HTTPServer, part of the CocoaHTTPServer sample code
+//  http://developer.apple.com/samplecode/CocoaHTTPServer/index.html
```
Comment (paulb777) [inline Example/Core/Tests/GTMHTTPServer.h]: This link doesn't work. What's the licensing on the code?

File: Example/Core/Tests/GTMHTTPServer.m [hunk 4]
```diff
@@ -0,0 +1,631 @@
+/* Copyright (c) 2010 Google Inc.
```
Comment (paulb777) [inline Example/Core/Tests/GTMHTTPServer.m]: Is this really from 2010? Remove the "(c)" to be consistent with Firebase copyrights and pass travis

File: Example/Core/Tests/GTMHTTPServer.h [hunk 5]
```diff
@@ -0,0 +1,145 @@
+/* Copyright (c) 2010 Google Inc.
```
Comment (paulb777) [inline Example/Core/Tests/GTMHTTPServer.h]: Is this really from 2010? Remove the "(c)" to be consistent with Firebase copyrights and pass travis

File: Example/Core/Tests/GTMHTTPServer.m [hunk 6]
```diff
@@ -0,0 +1,631 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//
+//  Based a little on HTTPServer, part of the CocoaHTTPServer sample code
+//  http://developer.apple.com/samplecode/CocoaHTTPServer/index.html
```
Comment (paulb777) [inline Example/Core/Tests/GTMHTTPServer.m]: linke doesn't work - what's the licensing?

**Comment by paulb777** [comment]
How about getting https://cocoapods.org/pods/gtm-http-fetcher from CocoaPods instead of copying the files?  Add a dependency in Example/Podfile

File: Example/Core/Tests/GTMHTTPServer.m (hunk 6 above)
Comment (morganchen12) [reply Example/Core/Tests/GTMHTTPServer.m]: Seems to be a BSD of some kind https://opensource.apple.com/source/HTTPServer/HTTPServer-11/OpenSourceLicenses/CocoaHTTPServer.txt.auto.html

**Comment by paulb777** [comment]
Because of the BSD derivation, the two files should go into a third_party directory with the BSD license inserted into the files.

**Comment by ryanwilson** [comment]
Moved the files into a `third_party` directory and added the BSD license inline - or should it be the entire file?

**Comment by paulb777** [comment]
License organization looks good to me.

**Comment by paulb777** [comment]
Merge or rebase master and do another travis run  - and then good to go!

**Comment by ryanwilson** [comment]
Finally got the rebase working, will wait for travis run.

**Pull request merged**